### DISCUSSION TITLE
Can `content-type` be set to header of response in `hono/streaming`?

### DISCUSSION THREAD

**Discussion by Kanahiro** [honojs/hono#2617] [Q&A]
`hono/stream` is good and I am trying to utilize this to response binary data.

However, `content-type` seems to be set to `text/plain` implicitly.
I've read codes to handle this and custom header looks not supported. Is it right? or is there some way to do?
https://github.com/honojs/hono/blob/cb2416042cea89419928aa21caf97de0edcc11e5/src/helper/streaming/stream.ts#L24

**Comment by yusukebe** [comment]
Hi @Kanahiro 

@sor4chi @nakasyou Any thoughts?

**Comment by Kanahiro** [comment]
While discussion but I pushed a proposal to handle this.

https://github.com/honojs/hono/pull/2619

If there is another way, I'm happy to be able to see it.

**Comment by sor4chi** [comment]
Hi, @Kanahiro @yusukebe 

I think the current API already allows you to set Content-Type as you like, but is this one out of context?

![Screenshot 2024-05-06 at 11 31 12](https://github.com/honojs/hono/assets/80559385/09a2d5b9-1997-43f5-ad70-36dba27b6115)

**Reply by Kanahiro** [reply]
wow, this is just what for me! thanks!

**Reply by yusukebe** [reply]
Yes. If you want to set the content type as you like, you can use `c.header()`.

@sor4chi Thanks!

**Reply by shiny** [reply]
Note: `c.header` must be set before the stream starts; once output begins, headers can no longer be modified.
